syntax = "proto3";

package leaderboard;

service LeaderBoard { // Interface exported by the server.
  rpc AuthenticateUser (LoginPassword) returns (TokenAuth) {}

  rpc RecordPlayerScore(stream PlayerScore) returns (stream ScoreResponse) {}

  rpc GetLeaderBoard(GetLB) returns (LeaderBoardResponse) {}
}

message LoginPassword {
  string login = 1;
  string password = 2;
}

message TokenAuth {
  string token = 1;
}

message PlayerScore {
  string name = 1;
  int64 score = 2;
}

message ScoreResponse {
  int64 rank = 1;
}

message GetLB {
  int32 page = 1;
  string name = 2;
}

message LeaderBoardRecord {
  string name = 1;
  int64 score = 2;
  int32 rank = 3;
}


message LeaderBoardResponse {
  repeated LeaderBoardRecord results = 1;
  repeated LeaderBoardRecord around_me = 2;
  int32 next_page = 3;
}